pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -217,7 +218,10 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
       requestStatistics.setErrorCode(QueryException.PQL_PARSING_ERROR_CODE);
       return new BrokerResponseNative(QueryException.getException(QueryException.PQL_PARSING_ERROR, e));
     }
-    PinotQuery pinotQuery = brokerRequest.getPinotQuery();
+
+    BrokerRequest serverBrokerRequest = GapfillUtils.stripGapfill(brokerRequest);
```

Comment:
(MAJOR) Let's set the query options first (some options are already set during the query compilation), then strip the gapfill. You may just set the stripped query options to be the original query options without making a copy

Reply:
It is done inside GapfillUtils.stripGapfill method.